src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -46,102 +38,125 @@ public async Task InvokeAsync(HttpContext context)
+                : await userContextBuilder.BuildUserContext(context).ConfigureAwait(false);
 
             var executer = context.RequestServices.GetRequiredService<IGraphQLExecuter<TSchema>>();
-            var token = GetCancellationToken(context);
 
-            // normal execution with single graphql request
-            if (gqlBatchRequest == null)
+            // Normal execution with single graphql request
+            if (bodyGQLBatchRequest == null)
             {
                 var stopwatch = ValueStopwatch.StartNew();
-                var result = await executer.ExecuteAsync(
-                    gqlRequest.OperationName,
-                    gqlRequest.Query,
-                    gqlRequest.GetInputs(),
-                    userContext,
-                    token).ConfigureAwait(false);
+                var result = await ExecuteRequestAsync(gqlRequest, userContext, executer, cancellationToken);
 
                 await RequestExecutedAsync(new GraphQLRequestExecutionResult(gqlRequest, result, stopwatch.Elapsed));
 
-                await WriteResponseAsync(context, writer, result).ConfigureAwait(false);
+                await WriteResponseAsync(context, writer, cancellationToken, result).ConfigureAwait(false);
             }
-            // execute multiple graphql requests in one batch
+            // Execute multiple graphql requests in one batch
             else
             {
-                var executionResults = new ExecutionResult[gqlBatchRequest.Length];
-                for (int i = 0; i < gqlBatchRequest.Length; ++i)
+                var executionResults = new ExecutionResult[bodyGQLBatchRequest.Length];
+                for (int i = 0; i < bodyGQLBatchRequest.Length; ++i)
                 {
-                    var request = gqlBatchRequest[i];
+                    var request = bodyGQLBatchRequest[i];
 
                     var stopwatch = ValueStopwatch.StartNew();
-                    var result = await executer.ExecuteAsync(
-                        request.OperationName,
-                        request.Query,
-                        request.GetInputs(),
-                        userContext,
-                        token).ConfigureAwait(false);
+                    var result = await ExecuteRequestAsync(request, userContext, executer, cancellationToken);
 
                     await RequestExecutedAsync(new GraphQLRequestExecutionResult(gqlRequest, result, stopwatch.Elapsed, i));
 
                     executionResults[i] = result;
                 }
 
-                await WriteResponseAsync(context, writer, executionResults).ConfigureAwait(false);
+                await WriteResponseAsync(context, writer, cancellationToken, executionResults).ConfigureAwait(false);
             }
         }
 
+        private static Task<ExecutionResult> ExecuteRequestAsync(GraphQLRequest gqlRequest, IDictionary<string, object> userContext, IGraphQLExecuter<TSchema> executer, CancellationToken token)
```

Comment:
It seems we can inline this method. At your discretion.

Reply:
Seems to still have two uses though, so I don't think it can be?

Reply:
This method just wraps a call to another method that is not much longer than the first. Therefore, it is not of much use, but there is no particular harm.

Reply:
Ahh, I see, yea I did it to clean up the call site a bit. Just spotted a bug with the `RequestExecutedAsync` call passing the wrong ref in the batch loop. Fixed that.